### DISCUSSION TITLE
How to add an extension dependency to another extension?

### DISCUSSION THREAD

**Discussion by mcanouil** [quarto-dev/quarto-cli#1866] [Q&A] [answered]
The extension [social-share](https://github.com/schochastics/quarto-social-share) by @schochastics depends on Font Awesome.
While there is a [fontawesome](https://github.com/quarto-ext/fontawesome) extension, it would be better to avoir conflict between the two, thus allowing to use them both, for `social-share` to depends on `fontawesome`.

How can we achieve this?

**Comment by jjallaire** [comment, marked as answer]
There are purposely not cross extension dependencies so that we don't need to implement a package manager / solver) to manage version conflicts (imagine your extension installs an update of fontawesome that breaks the user's existing version!). If you want to "depend on" another extension's functionality you need to embed its source code.

Note that formats often want to embed filters/shortcodes so we do have a special utility for that (the version used in the format is private and won't overwrite other versions of the same extension in use): https://quarto.org/docs/extensions/formats.html#extension-embedding

**Reply by mcanouil** [reply]
Ok, so when adding html dependency, the version has to be set properly in order for this overwrite safety to work.
Thank you!
